ocd/src/collection/cores/m0.rs

```diff
@@ -312,6 +377,38 @@ impl Core for M0 {
         Ok(())
     }
 
+    fn reset_and_halt(&self, mi: &mut MasterProbe) -> Result<(), DebugProbeError> {
```

Comment:
This is quite some additional stuff to do. I wonder how the flashing process even worked without beforehand.

Also, do you have an idea why all the different reset functionality is required?

Reply:
It would be good to write some comments documenting this

Reply:
Yes. I hope we can make an effort soon to document this stuff quite a bit. This was the initial idea on probe-rs too: Document stuffe extensively. But you know how it goes ...

Reply:
Yes, I probably should have documented this change a bit better.